### DISCUSSION TITLE
Move image displayed with --kitty

### DISCUSSION THREAD

**Discussion by tdtooke** [fastfetch-cli/fastfetch#442] [Q&A] [answered]
First the title for my question is wrong.  It was for an earlier question that I found the answer for.  I did a modified version of your example number 2 and I've noticed that if I don't use kitty to display my png log and just have the ascii logo then the information gets printed out on the first line.  If I use --kitty however the information gets printed on the second line.  It's not really a real problem but I am a neat freak.  I've looked through fastfetch --help and don't see a parameter for that.

with --kitty
![gratuitousfastfetch](https://user-images.githubusercontent.com/72276499/230708910-db733efe-14d3-442e-a3b1-ce0554afe812.png)

with regular logo
![secondshot](https://user-images.githubusercontent.com/72276499/230708996-7e44b09e-d31a-4d36-aa3d-bdf1a7e53786.png)

edit:  I've been experimenting.  The png I used was pretty huged so I scaled it to the size needed and that sorta fixed it, but now the information starts on the same line as the fastfetch command was issued on.  Visually it looks good but technically it doesn't feel right.  Not too worried about it at this point but if anybody knows how to make it behave identically whether or not you're using ascii or an image I'd love to hear it.;

**Comment by CarterLi** [comment]
Please upload the image you were using.

**Comment by tdtooke** [comment]
I don't have the original icon I used anymore but here's the resized one I use:
![fastarch](https://user-images.githubusercontent.com/72276499/230739436-1cd42f82-249c-48d3-87b4-2431b8b6c8b7.png)

To really give you an idea about what I mean now.  First I ran df then I ran fastarch.  You can see how the information gets printed at the top even though stuff from df is still there.  It doesn't do that if I use the ascii logo
![newsot](https://user-images.githubusercontent.com/72276499/230739498-79a94d83-6363-46f7-8569-a75ca232a0cf.png)

**Comment by LinusDierheimer** [comment, marked as answer]
Just to give a little insight on this:

The way fastfetch draws is (and must be) the following:
* Draw the logo left to right, top to bottom.
* Go x lines up, where x is the height of the logo
* Go as left as possible
* For each line go y right, where y is the width of the logo, then one line down

For this to work properly, fastfetch must know the height of the logo in lines. This is no problem for ascii arts, but tricky for images, as their size is measured in pixels. Terminals provide the following APIs to interop:
* Width and Height of the terminal window in characters
* Width and Height of the terminal in pixels

Fastfetch determines the height of the logo, by calculating `imagePixelHeight / (terminalPixelHeight / terminalCharacterHeight`. This is where inaccuracy comes into play. For example how to handle window heights, where the pixel height isn't a multiple of the character pixel height?

Neofetch solves this issue by clearing the whole screen and just going up as much as possible. I didn't want to have such side effects, so i tried my best to calculate the correct height as good as possible for all terminals, image types, graphic protocols, image dimensions, etc.

If the calculation is wrong, usually only by a delta of one, going 3 to high isn't usual and should be investigated.

The easiest way to quickly work around this is to set an explicit logo height using `--logo-height`. You can also try experimenting with the `--logo-padding-top` option.

**Comment by tdtooke** [comment]
Thank you!!  I really didn't want to read the code because for images I suspect it'd take a lot longer for me to understand it.
Edit: --logo-height fixes everything!
